tests/rules/prefer-object-literal.test.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * eslint-plugin-sonarjs
+ * Copyright (C) 2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import { RuleTester } from "eslint";
+
+const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 2018 } });
+import rule = require("../../src/rules/prefer-object-literal");
+
+ruleTester.run("prefer-literal", rule, {
+  valid: [
+    {
+      code: `var x = {a : 2}`,
+    },
+    {
+      code: `
+      function Foo(a) { 
+        this.a = a; 
+      };
+      var x = new Foo(2);`,
+    },
+    {
+      code: `
+      var x = {a : 2};
+      y = "foo";`,
+    },
+    // FN
+    {
+      code: `
+      var x;
+      x = {};
+      x.a = 2`,
+    },
+    // FN
+    {
+      code: `var x = {a : 2}; doSomething(); x.b = 3;`,
+    },
+    {
+      code: `
+      function foo() { 
+        var x = {a : 2};
+        doSomething();
+      }`,
+    },
+    {
+      code: `var x = {}; x["a"] = 2;`,
+    },
+    // No Issue on function declaration, may be done for readibility
+    {
+      code: `
+      var x = {};
+      x.foo = function () {}`,
+    },
+    // OK, report only when empty object
+    {
+      code: `var x = {a : 2}; x.b = 5;`,
+    },
+  ],
+  invalid: [
+    {
+      code: `var x = {}; x.a = 2;`,
+      errors: [
+        {
+          message: "Convert this declaration to object literal syntax.",
+          line: 1,
+          endLine: 1,
+          column: 5,
+          endColumn: 11,
+        },
+      ],
+    },
+    {
+      code: `var x = {}, y = "hello"; x.a = 2;`,
```

Comment:
I don't understand why and also how we raise an issue here

Reply:
My intention is testing highlighting in multiple variable declaration.
```javascript
var x = {}, // Noncompliant
      y = "hello";
x.a = 2;
```